core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -671,7 +674,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
       classOf[LongWritable],
       classOf[BytesWritable],
       conf=conf)
-    val data = br.map{ case (k, v) => v.getBytes}
+    val data = br.map { case (k, v) =>
+      val bytes = v.getBytes
+      assert(bytes.length == recordLength, "Byte array does not have correct length")
+      bytes
```

Comment:
Gotcha, I think it's ok as is then. Given what `FixedLengthInputFormat` is doing, this is more a defensive assertion, it's not something the user should hit due to an inappropriate input.